common/util.go

```diff
@@ -265,3 +265,36 @@ func (s *ServiceClient) DatalakeServiceClient() (*datalake.Client, error) {
 	}
 	return s.dsc, nil
 }
+
+// Metadata utility functions to work around GoLang's metadata capitalization
+
+func TryAddMetadata(metadata Metadata, key, value string) {
+	if _, ok := metadata[key]; ok {
+		return // Don't overwrite the user's metadata
+	}
+
+	if key != "" {
+		capitalizedKey := strings.ToUpper(string(key[0])) + key[1:]
```

Comment:
Generally to make anything case insensitive we just convert everything to lower case and compare. Why we want to just capitalise the first character and keep the remaining as is. Anyway on blob you can not have two meta with same name with just some case change. Code will look more readable if we go that way,

Reply:
Iterating through the metadata is O(n) whereas adjusting our lookups is O(1) so I figured this is slightly better to just search manually.

We don't necessarily care about the full case insensitivity, just what Golang forces the metadata to look like.